usr/src/uts/common/fs/zfs/sys/spa.h

```diff
@@ -326,7 +327,8 @@ typedef enum bp_embedded_type {
 typedef struct blkptr {
 	dva_t		blk_dva[SPA_DVAS_PER_BP]; /* Data Virtual Addresses */
 	uint64_t	blk_prop;	/* size, compression, type, etc	    */
-	uint64_t	blk_pad[2];	/* Extra space for the future	    */
+	uint32_t	blk_iv[3];	/* IV for block level encryption    */
+	uint32_t	blk_pad;	/* Extra space for the future	    */
```

Comment:
> I'd much prefer to preserve the pad for when we really need it, and instead use the 3rd DVA.

@richardelling mentioned on the [original github issue](https://github.com/zfsonlinux/zfs/issues/494#issuecomment-180041652) that the 3rd DVA was being used by Nexenta for some acceleration. (I did not confirm this, but it seemed like enough reason to avoid putting it there.)

> I'm told that it may be possible (and still secure) to use 2 of the checksum words for the IV. That would be even better than consuming the 3rd DVA.

I'm already using 2 words of the checksum for the MAC. I could put the IV in the remaining 2 words, but then we really don't use any checksumming algorithm at all.

> Also, this needs to remain uint64_t's, so that it will be byteswapped correctly.

OK. I can do this, although it will mean that half of the 2nd IV word is unused, which is a bit strange.